cpu/cc2538/ldscripts/cc2538sf53.ld

```diff
@@ -15,13 +15,18 @@
  *
  * @author          Hauke Petersen <hauke.petersen@fu-berlin.de>
  */
+ 
+_rom_offset = DEFINED( _rom_offset ) ? _rom_offset : 0x0;
+_fw_rom_length = DEFINED( _fw_rom_length ) ? _fw_rom_length : _rom_length - _rom_offset;
+
+ASSERT((_fw_rom_length <= _rom_length - _rom_offset), "Specified firmware size does not fit in ROM");
 
 /* Memory Space Definitions: */
 MEMORY
 {
-    rom (rx)    : ORIGIN = 0x00200000, LENGTH = 512K - 44
+    rom (rx)    : ORIGIN = _rom_start_addr + _rom_offset, LENGTH = _fw_rom_length
     cca         : ORIGIN = 0x0027ffd4, LENGTH = 44
```

Comment:
This is a problem. when saying that cca will start at `0x0027ffd4` you will have this linked for both files, so what happens is that when flashing `slot0` you are completely erasing `slot1`, you can see this by looking at the generated binaries (on my local computer.)

```
-rw-rw-r--  1 francisco francisco  505008 Jun 18 18:09 tests_riotboot.map
-rwxrwxr-x  1 francisco francisco  519936 Jun 18 18:09 tests_riotboot-slot0.bin
-rwxrwxr-x  1 francisco francisco  599032 Jun 18 18:09 tests_riotboot-slot0.elf
-rw-rw-r--  1 francisco francisco     256 Jun 18 18:09 tests_riotboot-slot0.hdr
-rw-rw-r--  1 francisco francisco  520192 Jun 18 18:09 tests_riotboot-slot0.riot.bin
-rwxrwxr-x  1 francisco francisco  259840 Jun 18 18:09 tests_riotboot-slot1.bin
-rwxrwxr-x  1 francisco francisco  599032 Jun 18 18:09 tests_riotboot-slot1.elf
```

Reply:
Oops, I hadn't read correctly in the description that you had disabled it manually.

Reply:
Small question, what do you mean exactly by disabling the cca region? Do you comment out the section and memory definition? Could you paste the modified ld script? :)

Reply:
Hi, i just commented out "SECTIONS"
`
_rom_offset = DEFINED( _rom_offset ) ? _rom_offset : 0x0;
_fw_rom_length = DEFINED( _fw_rom_length ) ? _fw_rom_length : _rom_length - _rom_offset;
ASSERT((_fw_rom_length <= _rom_length - _rom_offset), "Specified firmware size does not fit in ROM");
MEMORY
{
    rom (rx)    : ORIGIN = _rom_start_addr + _rom_offset, LENGTH = _fw_rom_length
    cca         : ORIGIN = 0x0027ffd4, LENGTH = 44
    ram (w!rx)  : ORIGIN = _ram_start_addr,               LENGTH = _ram_length
}
/* MCU Specific Section Definitions */
**/*
SECTIONS
{
    .flashcca :
    {
        KEEP(*(.flashcca))
    } > cca
}
*/**
INCLUDE cortexm_base.ld`

Reply:
I think I have similar HW as you do, I have a cc2538em + smartrf06eb board. I'll try to replicate your setup (never used any of these board or TI software so it might take me some time to get back to you). What exact hardware are you using? Just to have an Idea of the differences we might have (if any)

Reply:
Hi, I am using a SmartRF06 Evaluation Board, with a CC2538EM 1.2.